testsuite/integration-arquillian/tests/other/console/src/test/java/org/keycloak/testsuite/console/BasicConsoleTest.java

```diff
@@ -37,4 +44,29 @@ public void testPostWindowMessage() throws InterruptedException {
         assertEquals("Expected window not to have reloaded", "check", result);
     }
 
+    @Test
+    // KEYCLOAK-17387
+    public void testUserWithViewUsersRoleCanOpenConsole() {
+        UserRepresentation userRepresentation = createTestUserWithViewUsersRole();
+        try {
+            loginToTestRealmConsoleAs(userRepresentation);
```

Comment:
I believe we somehow need to assert that the console works, that it's not at the Forbidden screen.

Reply:
You are right. My first assumption was, that the assertions of the method "loginToTestRealmConsoleAs" will be enough. But after a deeper look inside this is not the case.
I added a explicit check for that now.

Reply:
Thanks for the update!

I think we should do this in a more robust way. If some AJAX is still processing and we fetch the current URL, it might be unchanged. We should use the helper methods we have for this and ideally also create a Forbidden Page Object. Something like this:
https://github.com/keycloak/keycloak/blob/005801126570d857338e23198c65069a1ab2455e/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/OfflineServletsAdapterTest.java#L106

Sorry to bother you with this. :) UI tests can be sometimes tricky.

Reply:
Thank you for your feedback.
No worries :) Yeah UI tests are tricky.
I added a Forbidden page object and used the already existing assertion method. Is this the way you suggested?